tests/integration/cmd_dev_test.go

```diff
@@ -548,153 +560,168 @@ ComponentSettings:
+						helper.Cmd("odo", "init", "--name", cmpName, "--devfile-path", helper.GetExamplePath("source", "devfiles", "nodejs", "devfile-with-multiple-endpoints.yaml")).ShouldPass()
 					})
 
-					AfterEach(func() {
-						devSession.Stop()
-						devSession.WaitEnd()
-					})
-
-					It("should expose all endpoints on localhost regardless of exposure", func() {
-						getServerResponse := func(p int) string {
-							resp, err := http.Get(fmt.Sprintf("http://%s", ports[strconv.Itoa(p)]))
+					When("running odo dev", func() {
+						var devSession helper.DevSession
+						var ports map[string]string
+						BeforeEach(func() {
+							opts := []string{}
+							if manual {
+								opts = append(opts, "--no-watch")
+							}
+							var err error
+							devSession, _, _, ports, err = helper.StartDevMode(helper.DevSessionOpts{
+								CmdlineArgs: opts,
+								RunOnPodman: podman,
+							})
 							Expect(err).ToNot(HaveOccurred())
-							defer resp.Body.Close()
-
-							body, _ := io.ReadAll(resp.Body)
-							return string(body)
-						}
-						containerPorts := []int{3000, 4567, 7890}
+						})
 
-						for _, p := range containerPorts {
-							By(fmt.Sprintf("exposing a port targeting container port %d", p), func() {
-								r := getServerResponse(p)
-								helper.MatchAllInOutput(r, []string{"Hello from Node.js Starter Application!"})
-							})
-						}
+						AfterEach(func() {
+							devSession.Stop()
+							devSession.WaitEnd()
+						})
 
-						helper.ReplaceString("server.js", "Hello from Node.js", "H3110 from Node.js")
+						It("should expose all endpoints on localhost regardless of exposure", func() {
+							getServerResponse := func(p int) string {
+								resp, err := http.Get(fmt.Sprintf("http://%s", ports[strconv.Itoa(p)]))
+								Expect(err).ToNot(HaveOccurred())
+								defer resp.Body.Close()
 
-						if manual {
-							if os.Getenv("SKIP_KEY_PRESS") == "true" {
-								Skip("This is a unix-terminal specific scenario, skipping")
+								body, _ := io.ReadAll(resp.Body)
+								return string(body)
 							}
+							containerPorts := []int{3000, 4567, 7890}
 
-							devSession.PressKey('p')
-						}
-
-						_, _, _, err := devSession.WaitSync()
-						Expect(err).Should(Succeed())
-
-						for _, p := range containerPorts {
-							By(fmt.Sprintf("returning the right response when querying port forwarded for container port %d", p),
-								func() {
-									Eventually(func() string {
-										return getServerResponse(p)
-									}, 180, 10).Should(Equal("H3110 from Node.js Starter Application!"))
+							for _, p := range containerPorts {
+								By(fmt.Sprintf("exposing a port targeting container port %d", p), func() {
+									r := getServerResponse(p)
+									helper.MatchAllInOutput(r, []string{"Hello from Node.js Starter Application!"})
 								})
-						}
+							}
+
+							//						helper.ReplaceString("server.js", "Hello from Node.js", "H3110 from Node.js")
+							//
```

Comment:
Was this whole block commented out intentionnally? I think it is worth testing this case (relevant when Devfile has multiple endpoints), no? But we could make it conditional as manual sync is not supported with Podman mode for now..

Reply:
No, I changed it to an if podman condition instead. Thanks